ci/functional/test_main.sh

```diff
@@ -50,8 +55,13 @@ clush -B -S -o '-i ci_key' -l root -w "${first_node}" \
 
 if ! test_cluster; then
     # Sometimes a cluster reboot will fix the issue so try it once.
-    cluster_reboot
-    test_cluster
+    if cluster_reboot; then
+        if test_cluster; then
+            hardware_ok=true
+        fi
+    fi
+else
+    hardware_ok=true
 fi
```

Comment:
Not really a big deal (i.e. no reason to push a new commit just for this, but if you had to push a commit for something else, perhaps worth addressing) but this can be factored down to:

```suggestion
# Sometimes a cluster reboot will fix the issue so try it once.
if test_cluster || { cluster_reboot &&  test_cluster; }; then
    hardware_ok=true
fi
```